app/src/main/java/com/learner/learndroid/view/DetailActivity.java

```diff
@@ -0,0 +1,139 @@
+                .build();
+        WalmartService service = retrofit.create(WalmartService.class);
+        ProductRepository productRepository = new ProductRepository(service);
+        detailViewModel.setRepository(productRepository);
+
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        String itemId = getIntent().getStringExtra(PRODUCT_ID);
+        detailViewModel.getItem(itemId).observe(this, new Observer<Item>() {
+            @Override
+            public void onChanged(@Nullable Item item) {
+                updateDetailView(item);
+            }
+        });
+    }
+
+    /**
+     * Updates the detail view.
+     *
+     * @param item Product object.
+     */
+    private void updateDetailView(Item item) {
+        Log.d(TAG, "Product Received.");
+        if (item == null)
+            throw new NullPointerException("Product Detail Unavailable.");
+        ImageView productImageView = findViewById(R.id.product_detail_image);
+        RatingBar ratingView = findViewById(R.id.ratingView);
+        TextView ratingText = findViewById(R.id.ratingText);
+        TextView productName = findViewById(R.id.detail_product_name);
+        TextView productOriginalPrice = findViewById(R.id.product_original_price);
+        TextView productDealPrice = findViewById(R.id.product_sale_price);
+        TextView productDescription = findViewById(R.id.product_detail_description);
+        Button doneButton = findViewById(R.id.done_button);
+
+        Glide.with(this)
+                .asBitmap()
+                .load(item.getLargeImage())
+                .into(productImageView);
+
+        productName.setText(item.getName());
+
+
+        Double msrp = 0d;
+        Double sale = 0d;
+
+        if(item.getMsrp() != null) {
+            msrp = item.getMsrp();
+        }
+
+        if(item.getSalePrice() != null) {
+            sale = item.getSalePrice();
+        }
+
+        String originalPrice = String.format(Locale.US, "%.2f", msrp);
+        String originalPriceText = getApplicationContext().getString(R.string.original_price_text, originalPrice);
+        productOriginalPrice.setText(originalPriceText);
+
+        String dealPrice = String.format(Locale.US, "%.2f", sale);
+        String dealPriceText = getApplicationContext().getString(R.string.deal_price_text, dealPrice);
+        productDealPrice.setText(dealPriceText);
+
+        productDescription.setText(item.getLongDescription());
+
+        float productRating = Float.valueOf(item.getCustomerRating());
+        String productRatingValue = String.valueOf(productRating);
+        String productRatingText = getApplicationContext().getString(R.string.product_rating_text, productRatingValue);
+        ratingView.setRating(productRating);
+        ratingText.setText(productRatingText);
+
+        doneButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
```

Comment:
Suspect: Are we not using ButterKnife?

Reply:
That will be coming in the subsequent pull requests.